datastore/transaction.go

```diff
@@ -67,6 +70,22 @@ func (w maxAttempts) apply(s *transactionSettings) {
 	}
 }
 
+// WithReadTime returns a TransactionOption that specifies a snapshot of the
+// database to view.
+func WithReadTime(t time.Time) TransactionOption {
```

Comment:
This pattern looks better to me and is more user friendly compared to requiring users to configure a struct correctly. Could we use this above?

Reply:
Done.